src/mono/mono/eglib/glib.h

```diff
@@ -69,6 +69,12 @@
 #error Mono requires _Noreturn (C11 or newer)
 #endif
 
+G_ATTR_NORETURN
+static inline void eg_unreachable (void) {
+	for (;;)
+		;
```

Comment:
I think this should use the builtins, when available:

```suggestion
#if defined(_MSC_VER)
	 __assume(0);
#elif defined(__GNUC__) && ((__GNUC__ > 4) || (__GNUC__ == 4 && (__GNUC_MINOR__ >= 5)))
	__builtin_unreachable();
#else
	for (;;)
	;
#endif
```